### PR TITLE
Convert to strict-manifest interface, remove compilation utils

### PR DESCRIPTION
**Pull request by njgheorghita**
## What was wrong?
`pte` had multiple deployers, that would either load from manifest, or compile assets (depending on the contract's language). In order to simplify `pte`'s api, it now only needs a manifest (EthPM spec-compliant) in order to create a `deployer` (regardless of the contract language). It now only exposes a single `deployer` fixture rather than `vy_deployer` & `sol_deployer`, and has no compilation abilities.

#### Cute Animal Picture
image

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -51,8 +51,8 @@
         "eth-abi>=1.2.2,<2",
         "eth-utils>=1,<2",
         "ethpm>=0.1.4a1,<1",
+        "twig>=0.1.0a5,<1",
```
Comment (pipermerriam) [inline setup.py]: This feels wrong.  `pytest-ethereum` is the low level library so I would expect it to know nothing about `twig` and instead for `twig` to install `pytest-ethereum` as a dependency.

Comment (njgheorghita) [reply setup.py]: I was under the impression that all compilation was going to happen in `twig`? In which case, `pytest-ethereum` would need access to the compilation utils from `twig` in order to create and feed the manifest into `pytest-ethereum`'s deployers. Or am I mistaken, and vyper contract compilation should happen in `pytest-ethereum`?

Comment (njgheorghita) [reply setup.py]: @pipermerriam We forgot to talk about this pr (and https://github.com/ethereum/twig/pull/11) in our 1-on-1. Still want to schedule a call to discuss?

Comment (pipermerriam) [reply setup.py]: I realize we still haven't resolved this.  Ping me and we can chat.

**Comment by njgheorghita** [comment]
@pipermerriam I'm having trouble figuring out what's causing circleci to fail - any thoughts?

**Comment by pipermerriam** [comment]
> `pip._vendor.packaging.requirements.InvalidRequirement: Parse error at "'(===3.2.'": Expected stringEnd`

That line makes me think that one of the requirements has an invalid version string.  I'd first try to reproduce locally in a new venv **or** re-run the circle-ci run with SSH enabled and SSH into the box there.  Probably will take a little digging to figure out which requirement it is.

**Comment by pipermerriam** [comment]
Looks like it was this: https://github.com/ethereum/py_ecc/pull/37

**Comment by njgheorghita** [comment]
@pipermerriam ping for review - removed all compilation and updated `pte`'s api to be manifest-based

File: docs/overview.rst [hunk 2]
```diff
@@ -95,21 +71,21 @@ For example, the `Escrow` contract factory requires linking to an instance of th
...
+   def escrow_deployer(deployer, manifest_dir):
        escrow_manifest_path = manifest_dir / "escrow_manifest.json"
-       return solc_deployer(escrow_manifest_path), w3
+       return deployer(escrow_manifest_path)
 
 
    @pytest.fixture
    def escrow_contract_instance(escrow_deployer):
-       deployer, w3 = escrow_deployer
+       w3 = escrow_deployer.package.w3
```
Comment (pipermerriam) [inline docs/overview.rst]: Seems like it would be better to get the `w3` instance from a fixture.

File: tests/core/test_deployer.py [hunk 3]
```diff
@@ -1,20 +1,13 @@
...
-from pytest_ethereum.deployer import Deployer
 from pytest_ethereum.exceptions import DeployerError
 
-logging.getLogger("evm").setLevel(logging.INFO)
-
-
-def test_deployer_fixture(request, vyper_project_dir):
-    deployer = request.getfixturevalue("vy_deployer")
-    assert isinstance(deployer, Deployer)
+logging.disable(level=logging.CRITICAL)
```
Comment (pipermerriam) [inline tests/core/test_deployer.py]: This deserves a comment to explain why... (why???)

**Comment by fubuloubu** [comment]
It seems like this update, `vyper` should be removed as a core dependancy?

**Pull request merged**